app/src/main/java/nextstep/shoppingcart/ui/screen/ProductItemScreen.kt

```diff
@@ -0,0 +1,76 @@
+package nextstep.shoppingcart.ui.screen
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.IntrinsicSize
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import coil.compose.AsyncImage
+import nextstep.shoppingcart.data.Product
+import java.text.NumberFormat
+import java.util.Locale
+
+@Preview(showBackground = true)
+@Composable
+private fun ProductItemPreview() {
+    val product = Product(
+        imageUrl = "https://picsum.photos/156/158",
+        name = "상품 이름을 테스트해보겠습니다 말줄입이 되나요",
+        price = 1200000000
+    )
+    ProductItem(product)
+}
+
+@Composable
+fun ProductItem(product: Product) {
+    Column(
+        modifier = Modifier
+            .width(156.dp)
+            .height(IntrinsicSize.Min)
+    ) {
+        AsyncImage(
+            model = product.imageUrl,
+            contentDescription = null,
+            modifier = Modifier
+                .width(156.dp)
+                .height(158.dp)
+        )
+        Column(
+            modifier = Modifier
+                .padding(top = 4.dp, start = 4.dp)
+                .fillMaxSize()
+        ) {
```

Comment:
> 해당 속성은 Column의 height를 유동적으로 변경하기 위해서 사용을 해본건데요. xml의 wrap_content느낌으로요. 무언가 잘못되었을까요?****

fillMaxSize()가 있어서 원래대로라면 이 Column이 전체를 채워야 합니다.
그런데 IntrinsicSize.Min 속성이 들어가면서 우연히 잘 나오게 된 것 같은데요.
높이에 대한 Modifier를 따로 설정하지 않으면, 자연스럽게 유동적으로 늘어나게 됩니다.

아래는 제가 생각한 배치 방법인데 참고 부탁드립니다.
```suggestion
    Column(modifier = modifier) {
        AsyncImage(
            model = product.imageUrl,
            contentDescription = null,
            modifier = Modifier.fillMaxWidth().aspectRatio(1f)
        )
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(top = 4.dp, start = 4.dp)
        ) {
```

```kotlin
ProductItem(
   product = product,
   modifier = Modifier.weight(1f)
)
```

Reply:
width()를 156으로 고정하면 생길 수 있는 문제점
> 폴드와 같은 화면에서는 오른쪽이 한참 남게 됨
> 만약 가로가 모자라다면 UI가 잘려 보일 수 있음

width()를 고정하게 되는 것에서 weight로 변경하게 되면
--> 아이템이 홀수로 떨어지게 되면 꽉차게 그려져서 이상한 문제가 됨

이는 LazyVerticalGrid를 적용 하거나, 오르쪽을 빈 영역으로 채우는 방법이 있겠습니다.

Reply:
네 언급주신 내용처럼 폴드와 같은 곳에서는 한참 남게 되겠죠. 사실 이런 경우 가운데 여백을 무게를 두고 나눌 것인지 아니면 이미지 자체를 큰 형식으로 보여줄 것인지에 대한 고민이 필요할 것 같은데요. 사실 피그마만 본 경우에 어떤식으로 구성을 해야 할지 좀 애매한 부분이 있더라고요. 말씀 주신 내용대로라면 weight만 주는 방식으로 해결이 된다로 이해를 하면 될까요? 아 그리고 위에 언급 주신 속성은 제거를 했습니다.

Reply:
여러가지 경우의 수를 고려했을 때 LazyVerticalGrid + weight를 이용한 전략이 가장 적합해 보여서 작성해봤어요.
정답은 물론 여러가지가 있을 수 있겠죠